File: Frontend-admin/src/types/categoria.ts
export interface Categoria {
  id: number
  nombre: string
  descripcion: string | null
  imagen_url: string | null
  parent_id: number | null
}

export interface CategoriaCreate {
  nombre: string
  descripcion?: string
  imagen_url?: string
  parent_id?: number
}

export interface CategoriaUpdate {
  nombre?: string
  descripcion?: string
  imagen_url?: string
  parent_id?: number
}

File: Backend/app/schemas/categoria.py
from pydantic import BaseModel, Field
from typing import Optional


class CategoriaCreate(BaseModel):
    nombre: str = Field(min_length=1, max_length=100)
    descripcion: Optional[str] = None
    imagen_url: Optional[str] = None
    parent_id: Optional[int] = None


class CategoriaUpdate(BaseModel):
    nombre: Optional[str] = Field(default=None, min_length=1, max_length=100)
    descripcion: Optional[str] = None
    imagen_url: Optional[str] = None
    parent_id: Optional[int] = None


class CategoriaRead(BaseModel):
    id: int
    nombre: str
    descripcion: Optional[str]
    imagen_url: Optional[str]
    parent_id: Optional[int]

    model_config = {"from_attributes": True}
